pkg/testing/integration/program.go

```diff
@@ -168,6 +168,8 @@ type ProgramTestOptions struct {
 	QueryCommandlineFlags []string
 	// RunBuild indicates that the build step should be run (e.g. run `yarn build` for `nodejs` programs)
 	RunBuild bool
+	// RunUpdateTest will ensure that updates to the package version can test for spurious diffs
+	RunUpdateTest bool
```

Comment:
Just to understand how this is used.  If you want to use this feature, you should specify the package you are going to test as a `dependency` in `package.json` (instead of a `peerDependency`, as we do today) (in the case of python, it would mean adding an entry to requirements.txt) and then also list the packages you want to replaces as `Dependencies` in the ProgramTestOptions struct?

Reply:
You are 100% correct here - this is the only way that we can have a mix and match of these tests. I didn't understand why we were using PeerDependency for this but it doesn't work when we use that

Is this acceptance for you?

Reply:
> I didn't understand why we were using PeerDependency for this but it doesn't work when we use that

I don't think we had great reasons.  The one things `peerDepencies` bought us over not listing it in `package.json` was that `npm` and friends would remind you that you needed to provide a version.

> Is this acceptance for you?

Yup, also this is the best way I could think of solving this.